here-naksha-lib-hub/src/main/java/com/here/naksha/lib/hub/storages/NHSpaceStorageWriter.java

```diff
@@ -76,46 +80,83 @@ public NHSpaceStorageWriter(
+      return executeWriteToCustomSpaces(wc, collectionId);
     }
   }
 
   private @NotNull Result executeWriteFeatures(final @NotNull WriteFeatures wf) {
     final String spaceId = wf.getCollectionId();
     logger.info("WriteFeatures Request against spaceId={}", spaceId);
     addSpaceIdToStreamInfo(spaceId);
-    if (virtualSpaces.containsKey(spaceId)) {
+    if (isDeleteSpaceRequest(wf, spaceId)) {
+      return executeDeleteSpace(wf);
+    } else if (virtualSpaces.containsKey(spaceId)) {
       // Request is to write to Naksha Admin space
-      return executeWriteFeaturesToAdminSpaces(wf);
+      return executeWriteToAdminSpaces(wf, spaceId);
     } else {
       // Request is to write to Custom space
-      return executeWriteFeaturesToCustomSpaces(wf);
+      return executeWriteToCustomSpaces(wf, spaceId);
     }
   }
 
-  private @NotNull Result executeWriteFeaturesToAdminSpaces(final @NotNull WriteFeatures<?, ?, ?> wf) {
+  private @NotNull Result executeWriteToAdminSpaces(
+      final @NotNull WriteRequest<?, ?, ?> wr, @NotNull String spaceId) {
     // Run pipeline against virtual space
-    final String spaceId = wf.getCollectionId();
     final EventPipeline pipeline = pipelineFactory.eventPipeline();
     final Result result = setupEventPipelineForAdminVirtualSpace(spaceId, pipeline);
-    if (!(result instanceof SuccessResult)) return result;
-    return pipeline.sendEvent(wf);
+    if (!(result instanceof SuccessResult)) {
+      return result;
+    }
+    return pipeline.sendEvent(wr);
   }
 
-  private @NotNull Result executeWriteFeaturesToCustomSpaces(final @NotNull WriteFeatures<?, ?, ?> wf) {
-    final String spaceId = wf.getCollectionId();
+  private @NotNull Result executeWriteToCustomSpaces(
+      final @NotNull WriteRequest<?, ?, ?> wr, @NotNull String spaceId) {
     final EventPipeline eventPipeline = pipelineFactory.eventPipeline();
     final Result result = setupEventPipelineForSpaceId(spaceId, eventPipeline);
-    if (!(result instanceof SuccessResult)) return result;
-    return eventPipeline.sendEvent(wf);
+    if (!(result instanceof SuccessResult)) {
+      return result;
+    }
+    return eventPipeline.sendEvent(wr);
+  }
+
+  private boolean isDeleteSpaceRequest(@NotNull WriteFeatures<?, ?, ?> wf, @NotNull String spaceId) {
+    return NakshaAdminCollection.SPACES.equals(spaceId)
+        && wf.features.size() == 1
+        && EWriteOp.DELETE.toString().equals(wf.features.get(0).getOp());
+  }
+
+  private @NotNull Result executeDeleteSpace(@NotNull WriteFeatures<?, ?, ?> deleteSpaceEntryReq) {
+    if (deleteSpaceEntryReq.features.get(0).getFeature() instanceof Space space) {
+      WriteXyzCollections deleteSpaceReq = new WriteXyzCollections().delete(new XyzCollection(space.getId()));
+      Result deleteSpaceRes = executeWriteCollections(deleteSpaceReq);
+      if (deleteSpaceRes instanceof SuccessResult) {
+        return executeWriteToAdminSpaces(deleteSpaceEntryReq, NakshaAdminCollection.SPACES);
```

Comment:
FYI - I would just use `deleteSpaceEntryReq.getCollectionId()` instead of hardcoded `NakshaAdminCollection.SPACES` , to logically avoid overriding original intent just by accident. But as I don't think of any use-case where it would be any different, it is not that critical and so no hard choice here.

Reply:
Ok, will do :)

Just to note:
The whole `executeDeleteSpace` is executed only when `isDeleteSpaceRequest` returned `true`.
The `isDeleteSpaceRequest` explicitly checks against `SPACES` collection